src/cli/setup/sign.rs

```diff
@@ -0,0 +1,187 @@
+use crate::blockdata::Block;
+use crate::cli::setup::index_of;
+use crate::cli::setup::traits::Response;
+use crate::crypto::multi_party_schnorr::LocalSig;
+use crate::crypto::multi_party_schnorr::SharedKeys;
+use crate::crypto::vss::Vss;
+use crate::errors::Error;
+use crate::net::SignerID;
+use crate::signer_node::BidirectionalSharedSecretMap;
+use crate::signer_node::SharedSecret;
+
+use bitcoin::{PrivateKey, PublicKey};
+use clap::{App, Arg, ArgMatches, SubCommand};
+use curv::arithmetic::traits::Converter;
+use curv::cryptographic_primitives::secret_sharing::feldman_vss::ShamirSecretSharing;
+use curv::cryptographic_primitives::secret_sharing::feldman_vss::VerifiableSS;
+use curv::elliptic::curves::traits::{ECPoint, ECScalar};
+use curv::{BigInt, FE, GE};
+use std::fmt;
+use std::str::FromStr;
+
+pub struct SignResponse {
+    local_sig: LocalSig,
+    public_key: PublicKey,
+}
+
+impl SignResponse {
+    fn new(local_sig: LocalSig, public_key: PublicKey) -> Self {
+        SignResponse {
+            local_sig: local_sig,
+            public_key: public_key,
+        }
+    }
+}
+
+impl Response for SignResponse {}
+
+impl fmt::Display for SignResponse {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "{:0>64}{:0>64}{:0>66}",
```

Comment:
I think it is enough as just `{}` for 3rd one. public keys should be formatted 32 bytes.